tests/unit/aws/test_connect.py

```diff
@@ -0,0 +1,58 @@
+from unittest.mock import MagicMock, patch
+
+from localstack.aws.connect import ExternalClientFactory, InternalClientFactory
+
+
+class TestClientFactory:
+    def test_internal_client_dto_is_registered(self):
+        factory = InternalClientFactory()
+        factory._session = MagicMock()
+
+        mock = factory("sns", "eu-central-1")
+        mock.meta.events.register.assert_called()
```

Comment:
nit: this function states that it tests that dto is registered for internal clients, but does not assert this fact. It only asserts that the `register` method was called _somehow_. Is it worth checking the function arguments?

```suggestion
        mock.meta.events.register.assert_called_with("before-call.*.*", handler=_handler_inject_dto_header)
```

This does tightly couple the test with implementation details which is not a great idea, however you can get the function calls and check that the `_handler_inject_dto_header` function is amongst them which is a looser constraint, but is pretty verbose.